plugins/newsql/newsql.c

```diff
@@ -1047,10 +1047,11 @@ static int newsql_write_response(struct sqlclntstate *c, int t, void *a, int i)
         return newsql_columns_fdb_push(c, a, i);
     case RESPONSE_DEBUG: return newsql_debug(c, a);
     case RESPONSE_ERROR: return newsql_error(c, a, i);
-    case RESPONSE_ERROR_ACCESS: return newsql_error(c, a, CDB2ERR_ACCESS);
-    case RESPONSE_ERROR_BAD_STATE: return newsql_error(c, a, CDB2ERR_BADSTATE);
-    case RESPONSE_ERROR_PREPARE: return newsql_error(c, a, CDB2ERR_PREPARE_ERROR);
-    case RESPONSE_ERROR_REJECT: return newsql_error(c, a, CDB2ERR_REJECTED);
+    case RESPONSE_ERROR_ACCESS: return newsql_error(c, a, CDB2__ERROR_CODE__ACCESS);
+    case RESPONSE_ERROR_APPSOCK_LIMIT: return newsql_error(c, a, CDB2__ERROR_CODE__APPSOCK_LIMIT);
+    case RESPONSE_ERROR_BAD_STATE: return newsql_error(c, a, CDB2__ERROR_CODE__BADSTATE);
+    case RESPONSE_ERROR_PREPARE: return newsql_error(c, a, CDB2__ERROR_CODE__PREPARE_ERROR);
+    case RESPONSE_ERROR_REJECT: return newsql_error(c, a, CDB2__ERROR_CODE__REJECTED);
```

Comment:
I think it would be good to update the cdb2api code to decouple these protobuf codes from the cdb2api codes so that things don't break if someone changes one or the other.

I think that would involve updating [`cdb2_convert_error_code`](https://github.com/bloomberg/comdb2/blob/f2968c1a2a7c039b6156dbea4395151e9ce90993/cdb2api/cdb2api.c#L2932) plus a few places where cdb2api checks if the server sent a cdb2api code.

We could also leave the client code conversion for later but clearly document that the protobuf codes need to be in sync with the cdb2api codes for now.

What do you think?

Reply:
->We agreed to document that protobuf codes need to be the same as cdb2api codes and update cdb2api later